dev/AppNotifications/AppNotificationContentBuilder/AppNotificationContentBuilder.idl

```diff
@@ -0,0 +1,133 @@
+        ButtonConfig AddArgument(String key, Int32 value);
+        ButtonConfig AddArgument(String key, Double value);
+        ButtonConfig AddArgument(String key, Single value);
+        ButtonConfig AddArgument(String key, Boolean value);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    }
+
+    [contract(AppNotificationsContract, 2)]
+    enum AdaptiveImageCrop
+    {
+        Default = 0, // Cropping behavior determined by renderer.
+        None = 1, // Image is not cropped.
+        Circle = 2, // Image is cropped to a circle shape.
+    };
+
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass ImageConfig
+    {
+        ImageConfig(Windows.Foundation.Uri imageUri);
+
+        // A description of the image, for users of assistive technologies.
+        String AlternateText{ get; };
+        ImageConfig SetAlternateText(String alternateText);
+
+        // An enum to describe how the image will be cropped.
+        AdaptiveImageCrop HintCrop{ get; };
+        ImageConfig SetHintCrop(AdaptiveImageCrop hintCrop);
+
+        // Override the app logo with custom image of choice that will be displayed on the AppNotification, otherwise it will be an inline image.
+        Boolean IsAppLogo{ get; };
+        ImageConfig SetIsAppLogo(Boolean isAppLogo);
+
+        // A value whether Windows is allowed to append a query string to the image URI supplied in the Tile notification.
+        Boolean AddImageQuery{ get; };
+        ImageConfig SetAddImageQuery(Boolean addImageQuery);
+
+        // Retrieves the XML content of the image.
+        String GetXml();
+    }
+
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass AppNotificationContentBuilder
+    {
+        AppNotificationContentBuilder();
+
+        // Adds a key(without value) to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        AppNotificationContentBuilder AddArgument(String key);
+
+        // Adds a key/value to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        [default_overload]
+        AppNotificationContentBuilder AddArgument(String key, String value);
+        AppNotificationContentBuilder AddArgument(String key, Boolean value);
+        AppNotificationContentBuilder AddArgument(String key, Int32 value);
+        AppNotificationContentBuilder AddArgument(String key, Double value);
+        AppNotificationContentBuilder AddArgument(String key, Single value);
+
+        // Adds text to the toast with a language and a maximum number of lines.
+        AppNotificationContentBuilder AddText(String text, Int32 hintMaxLines, String language);
+
+        // Text to be displayed as attribution text.
+        AppNotificationContentBuilder AddAttributionText(String text);
+
+        // Text to be displayed as attribution text with language tags such as "en-US" or "fr-FR".
```

Comment:
Is there any enum that we can leverage on to specify language? Or is it a common thing to pass that information as a string?